airflow/models/taskinstance.py

```diff
@@ -425,6 +425,16 @@ class TaskInstance(Base, LoggingMixin):
 
     dag_run = relationship("DagRun", back_populates="task_instances", lazy='joined', innerjoin=True)
     rendered_task_instance_fields = relationship("RenderedTaskInstanceFields", lazy='noload', uselist=False)
+    sla_miss = relationship(
```

Comment:
@ashb @uranusjr this is what i came up with for defining a "FK" relationship without actually defining a db-level FK constraint.  does this look right?

Reply:
Looks good to me but I’d just spell ou `primaryjoin` literally; using `join` seems like an overkiil.

```python
primaryjoin="""and_(
    TaskInstance.dag_id == foreign(SlaMiss.dag_id),
    TaskInstance.task_id == foreign(SlaMiss.task_id),
    TaskInstance.run_id == foreign(SlaMiss.run_id),
    TaskInstance.map_index == foreign(SlaMiss.map_index),
```

Reply:
But I think we will add the fk anyway? (Based on discussion on Slack)

Reply:
can do literal but we can't create FK here because SlaMiss doesn't always have a TI to key to